Detections/OfficeActivity/NRT_Office_MailForwarding.yaml

```diff
@@ -17,45 +17,21 @@ relevantTechniques:
 query: |
   OfficeActivity
   | where OfficeWorkload =~ "Exchange"
-  | where Parameters has "ForwardingSmtpAddress"
-  | extend parsed = parse_json(Parameters)
-  | mv-expand parsed
-  | where parsed.Name == "ForwardingSmtpAddress"
-  | extend parameterName = tostring(parsed.Name), fwdingDestination = tostring(parsed.Value)
-  | where isnotempty(fwdingDestination)
-  | extend ClientIPOnly = case(
-  ClientIP has "." and ClientIP has ':', tostring(split(ClientIP,":")[0]),
-  ClientIP has "." and ClientIP has '-', tostring(split(ClientIP,"-")[0]),
-  ClientIP has ']-', tostring(trim_start(@'[[]',tostring(split(ClientIP,"]")[0]))),
-  ClientIP has ']:', tostring(trim_start(@'[[]',tostring(split(ClientIP,"]")[0]))),
-  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has ':', tostring(split(ClientIP_,":")[0]),
-  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has '-', tostring(split(ClientIP_,"-")[0]),
-  isempty(ClientIP) and ClientIP_ has ']-', tostring(trim_start(@'[[]',tostring(split(ClientIP_,"]")[0]))),
-  isempty(ClientIP) and ClientIP_ has ']:', tostring(trim_start(@'[[]',tostring(split(ClientIP_,"]")[0]))),
-  isnotempty(ClientIP), ClientIP,
-  isnotempty(ClientIP_), ClientIP_,
-  "IP Not Available"
-  )
-  | extend Port = case(
-  ClientIP has "." and ClientIP has ':', tostring(split(ClientIP,":")[1]),
-  ClientIP has "." and ClientIP has '-', tostring(split(ClientIP,"-")[1]),
-  ClientIP has ']-', tostring(split(ClientIP,"]-")[1]),
-  ClientIP has ']:', tostring(split(ClientIP,"]:")[1]),
-  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has ':', tostring(split(ClientIP_,":")[1]),
-  isempty(ClientIP) and ClientIP_ has "." and ClientIP_ has '-', tostring(split(ClientIP_,"-")[1]),
-  isempty(ClientIP) and ClientIP_ has ']-', tostring(split(ClientIP_,"]-")[1]),
-  isempty(ClientIP) and ClientIP_ has ']:', tostring(split(ClientIP_,"]:")[1]),
-  isnotempty(ClientIP), ClientIP,
-  isnotempty(ClientIP_), ClientIP_,
-  "IP Not Available"
-  )
-  | extend UserId = iff(isempty(UserId), UserId_, UserId)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), DistinctUserCount = dcount(UserId), UserId = make_set(UserId),
-  Ports = make_set(Port), EventCount = count() by fwdingDestination, ClientIP = ClientIPOnly
+  | mv-apply DynamicParameters = todynamic(Parameters) on (summarize ParsedParameters = make_bag(pack(tostring(DynamicParameters.Name), DynamicParameters.Value)))
+  | evaluate bag_unpack(ParsedParameters, columnsConflict='replace_source')
+  | extend DestinationMailAddress = tolower(case(
```

Comment:
Given this is an NRT rule it would be good have a check early on to see if Parameters has any of ForwardTo, RedirectTo, or ForwardingSmtpAddress before parsing out to improve performance.

Reply:
@petebryan Added.